Fingerprinting a slow view in MySQL doesn't time out and retries on each sync

### Describe the bug

When fingerprinting a slow view/table in MySQL, due to [https://github.com/metabase/metabase/issues/59070](<https://github.com/metabase/metabase/issues/59070>), it will hang until getting killed by the server. Connection resets and the query is still left running on the server (this happened to a customer). Fingerprinting will fail to complete and it retries on next sync, repeating the cycle again and leaving more of those queries running on the server

### To Reproduce

1. Start a Metabase with env var `MB_DB_QUERY_TIMEOUT_MINUTES=1`
2. Add a MySQL 8.0, on it set the timeout to 65s `SET GLOBAL MAX_EXECUTION_TIME=65000`
3. Create a view that takes 2 minutes:

```sql
CREATE VIEW my_sleep_view AS
SELECT SLEEP(120) AS my_col
```

Our setup is: 1m timeout on Metabase, 1m5s timeout on the DB, 2m for the view - Metabase should be timing out<br>4. Sync the DB<br>5. Timeout isn't triggered by MB (see root issue mentioned in description above) so it is stopped by the DB, which means we failed to time out:

```
2025-11-17 11:11:58.868 | 2025-11-17 14:11:58,858 WARN sync.util :: Error fingerprinting Table 142 ''my_sleep_view'' 
2025-11-17 11:11:58.869 | clojure.lang.ExceptionInfo: Error executing query: (conn=440) Query execution was interrupted, maximum statement 
```

If you repeat the sync, it keeps failing to fingerprint here.

### Expected behavior

It should time out the fingerprint at least at `MB_DB_QUERY_TIMEOUT_MINUTES` or stop retrying this slow view on future syncs.

### Logs

```
2025-11-17 11:10:38.629 | 2025-11-17 14:10:38,629 INFO sync.util :: STARTING: Sync mysql Database 2 ''mysql-14020'' (free: 0.13G, total: 0.84G, ratio: 0.16) 
2025-11-17 11:10:38.629 | 2025-11-17 14:10:38,629 INFO sync.util :: STARTING: Sync metadata for mysql Database 2 ''mysql-14020'' (free: 0.13G, total: 0.84G, ratio: 0.16) 
2025-11-17 11:10:38.653 | 2025-11-17 14:10:38,653 DEBUG middleware.log :: POST /api/database 200 149ms (23 DB calls) App DB connections: 0/7 Jetty threads: 5/50 (5 idle, 0 queued) (62 total active threads) Queries in flight: 0 (0 queued); mysql DB 2 connections: 0/0 (1 threads blocked) {:metabase-user-id 1} 
2025-11-17 11:10:38.662 | 2025-11-17 14:10:38,662 INFO sql-jdbc.execute :: Obtaining a fresh resilient connection 
2025-11-17 11:10:41.074 | 2025-11-17 14:10:41,074 DEBUG middleware.log :: GET /api/session/properties 200 9ms (16 DB calls) App DB connections: 0/7 Jetty threads: 5/50 (5 idle, 0 queued) (63 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1} 
2025-11-17 11:10:41.102 | 2025-11-17 14:10:41,102 DEBUG middleware.log :: GET /api/user/current 200 5ms (12 DB calls) App DB connections: 0/7 Jetty threads: 5/50 (5 idle, 0 queued) (63 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1} 
2025-11-17 11:10:41.122 | 2025-11-17 14:10:41,121 DEBUG middleware.log :: GET /api/database 200 2ms (3 DB calls) App DB connections: 1/7 Jetty threads: 9/50 (3 idle, 0 queued) (63 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1} 
2025-11-17 11:10:41.122 | 2025-11-17 14:10:41,121 DEBUG middleware.log :: GET /api/database 200 2ms (3 DB calls) App DB connections: 0/7 Jetty threads: 9/50 (3 idle, 0 queued) (63 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1} 
2025-11-17 11:10:41.128 | 2025-11-17 14:10:41,128 DEBUG middleware.log :: GET /api/search 200 8ms (5 DB calls) App DB connections: 0/7 Jetty threads: 5/50 (4 idle, 0 queued) (64 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1} 
2025-11-17 11:10:43.147 | 2025-11-17 14:10:43,147 DEBUG middleware.log :: GET /api/database 200 5ms (3 DB calls) App DB connections: 0/7 Jetty threads: 5/50 (5 idle, 0 queued) (65 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1} 
2025-11-17 11:10:43.153 | 2025-11-17 14:10:43,153 DEBUG middleware.log :: GET /api/database 200 3ms (3 DB calls) App DB connections: 0/7 Jetty threads: 5/50 (5 idle, 0 queued) (65 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1} 
2025-11-17 11:10:45.174 | 2025-11-17 14:10:45,173 DEBUG middleware.log :: GET /api/database 200 4ms (3 DB calls) App DB connections: 0/7 Jetty threads: 6/50 (5 idle, 0 queued) (65 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1} 
2025-11-17 11:10:45.179 | 2025-11-17 14:10:45,179 DEBUG middleware.log :: GET /api/database 200 4ms (3 DB calls) App DB connections: 0/7 Jetty threads: 5/50 (5 idle, 0 queued) (65 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1} 
2025-11-17 11:10:47.199 | 2025-11-17 14:10:47,199 DEBUG middleware.log :: GET /api/database 200 5ms (3 DB calls) App DB connections: 0/7 Jetty threads: 5/50 (5 idle, 0 queued) (65 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1} 
2025-11-17 11:10:47.205 | 2025-11-17 14:10:47,205 DEBUG middleware.log :: GET /api/database 200 3ms (3 DB calls) App DB connections: 0/7 Jetty threads: 5/50 (5 idle, 0 queued) (65 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1} 
2025-11-17 11:10:49.225 | 2025-11-17 14:10:49,224 DEBUG middleware.log :: GET /api/database 200 4ms (3 DB calls) App DB connections: 0/7 Jetty threads: 6/50 (5 idle, 0 queued) (65 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1} 
2025-11-17 11:10:49.229 | 2025-11-17 14:10:49,228 DEBUG middleware.log :: GET /api/database 200 4ms (3 DB calls) App DB connections: 0/7 Jetty threads: 5/50 (5 idle, 0 queued) (65 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1} 
2025-11-17 11:10:51.250 | 2025-11-17 14:10:51,249 DEBUG middleware.log :: GET /api/database 200 6ms (3 DB calls) App DB connections: 1/7 Jetty threads: 6/50 (5 idle, 0 queued) (65 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1} 
2025-11-17 11:10:51.253 | 2025-11-17 14:10:51,253 DEBUG middleware.log :: GET /api/database 200 3ms (3 DB calls) App DB connections: 0/7 Jetty threads: 5/50 (5 idle, 0 queued) (65 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1} 
2025-11-17 11:10:53.276 | 2025-11-17 14:10:53,275 DEBUG middleware.log :: GET /api/database 200 4ms (3 DB calls) App DB connections: 1/7 Jetty threads: 6/50 (5 idle, 0 queued) (65 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1} 
2025-11-17 11:10:53.277 | 2025-11-17 14:10:53,277 DEBUG middleware.log :: GET /api/database 200 2ms (3 DB calls) App DB connections: 0/7 Jetty threads: 6/50 (5 idle, 0 queued) (65 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1} 
2025-11-17 11:10:53.697 | 2025-11-17 14:10:53,697 INFO sync.describe-database :: "my_sleep_view": SELECT privileges confirmed 
2025-11-17 11:10:53.701 | 2025-11-17 14:10:53,701 INFO sync.util :: STARTING: step ''sync-dbms-version'' for mysql Database 2 ''mysql-14020'' (free: 0.09G, total: 0.84G, ratio: 0.10) 
2025-11-17 11:10:53.716 | 2025-11-17 14:10:53,716 INFO sync.util :: FINISHED: step ''sync-dbms-version'' for mysql Database 2 ''mysql-14020'' (14.9 ms) (free: 0.08G, total: 0.84G, ratio: 0.10) 
2025-11-17 11:10:53.716 | 2025-11-17 14:10:53,716 INFO sync.util :: STARTING: step ''sync-timezone'' for mysql Database 2 ''mysql-14020'' (free: 0.08G, total: 0.84G, ratio: 0.10) 
2025-11-17 11:10:53.720 | 2025-11-17 14:10:53,720 INFO sync-metadata.sync-timezone :: :mysql database 2 default timezone is "UTC" 
2025-11-17 11:10:53.731 | 2025-11-17 14:10:53,730 INFO sync.util :: FINISHED: step ''sync-timezone'' for mysql Database 2 ''mysql-14020'' (13.9 ms) (free: 0.08G, total: 0.84G, ratio: 0.10) 
2025-11-17 11:10:53.731 | 2025-11-17 14:10:53,731 INFO sync.util :: STARTING: step ''sync-tables'' for mysql Database 2 ''mysql-14020'' (free: 0.08G, total: 0.84G, ratio: 0.10) 
2025-11-17 11:10:53.732 | 2025-11-17 14:10:53,732 INFO sync-metadata.tables :: Found new table: Table  ''my_sleep_view'' 
2025-11-17 11:10:53.738 | 2025-11-17 14:10:53,737 INFO sync-metadata.tables :: Updating table metadata for Table 142 ''my_sleep_view'' 
2025-11-17 11:10:53.738 | 2025-11-17 14:10:53,738 INFO sync-metadata.tables :: :is_writable of Table 142 ''my_sleep_view'' changed from null to true 
2025-11-17 11:10:53.739 | 2025-11-17 14:10:53,739 INFO sync.util :: FINISHED: step ''sync-tables'' for mysql Database 2 ''mysql-14020'' (8.5 ms) (free: 0.08G, total: 0.84G, ratio: 0.09) 
2025-11-17 11:10:53.739 | 2025-11-17 14:10:53,739 INFO sync.util :: STARTING: step ''sync-fields'' for mysql Database 2 ''mysql-14020'' (free: 0.08G, total: 0.84G, ratio: 0.09) 
2025-11-17 11:10:53.761 | 2025-11-17 14:10:53,760 INFO sync.util :: FINISHED: step ''sync-fields'' for mysql Database 2 ''mysql-14020'' (21.2 ms) (free: 0.07G, total: 0.84G, ratio: 0.09) 
2025-11-17 11:10:53.761 | 2025-11-17 14:10:53,761 INFO sync.util :: STARTING: step ''sync-fks'' for mysql Database 2 ''mysql-14020'' (free: 0.07G, total: 0.84G, ratio: 0.09) 
2025-11-17 11:10:53.773 | 2025-11-17 14:10:53,773 INFO sync.util :: FINISHED: step ''sync-fks'' for mysql Database 2 ''mysql-14020'' (12.0 ms) (free: 0.07G, total: 0.84G, ratio: 0.09) 
2025-11-17 11:10:53.773 | 2025-11-17 14:10:53,773 INFO sync.util :: STARTING: step ''sync-indexes'' for mysql Database 2 ''mysql-14020'' (free: 0.07G, total: 0.84G, ratio: 0.09) 
2025-11-17 11:10:53.774 | 2025-11-17 14:10:53,774 INFO sync.util :: FINISHED: step ''sync-indexes'' for mysql Database 2 ''mysql-14020'' (903.1 µs) (free: 0.07G, total: 0.84G, ratio: 0.09) 
2025-11-17 11:10:53.774 | 2025-11-17 14:10:53,774 INFO sync.util :: STARTING: step ''sync-metabase-metadata'' for mysql Database 2 ''mysql-14020'' (free: 0.07G, total: 0.84G, ratio: 0.09) 
2025-11-17 11:10:53.775 | 2025-11-17 14:10:53,774 INFO sync.util :: FINISHED: step ''sync-metabase-metadata'' for mysql Database 2 ''mysql-14020'' (754.8 µs) (free: 0.07G, total: 0.84G, ratio: 0.08) 
2025-11-17 11:10:53.781 | 2025-11-17 14:10:53,780 INFO sync.util :: FINISHED: Sync metadata for mysql Database 2 ''mysql-14020'' (15.1 s) (free: 0.07G, total: 0.84G, ratio: 0.08) 
2025-11-17 11:10:53.781 | 2025-11-17 14:10:53,781 INFO sync.util :: STARTING: Analyze data for mysql Database 2 ''mysql-14020'' (free: 0.07G, total: 0.84G, ratio: 0.08) 
2025-11-17 11:10:53.782 | 2025-11-17 14:10:53,782 INFO sync.util :: STARTING: step ''fingerprint-fields'' for mysql Database 2 ''mysql-14020'' (free: 0.07G, total: 0.84G, ratio: 0.08) 
2025-11-17 11:10:53.788 | 2025-11-17 14:10:53,787 INFO sync.analyze :: fingerprint-fields Analyzed [************······································] 😒   25% Table 142 ''my_sleep_view'' 
2025-11-17 11:10:53.788 | 2025-11-17 14:10:53,788 INFO analyze.fingerprint :: Fingerprinting 1 fields in table Table 142 ''my_sleep_view'' 
2025-11-17 11:10:55.297 | 2025-11-17 14:10:55,296 DEBUG middleware.log :: GET /api/database 200 7ms (3 DB calls) App DB connections: 1/7 Jetty threads: 6/50 (5 idle, 0 queued) (66 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1} 
2025-11-17 11:10:55.317 | 2025-11-17 14:10:55,316 DEBUG middleware.log :: GET /api/database 200 4ms (3 DB calls) App DB connections: 1/7 Jetty threads: 5/50 (5 idle, 0 queued) (66 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1} 
2025-11-17 11:11:58.868 | 2025-11-17 14:11:58,858 WARN sync.util :: Error fingerprinting Table 142 ''my_sleep_view'' 
2025-11-17 11:11:58.869 | clojure.lang.ExceptionInfo: Error executing query: (conn=440) Query execution was interrupted, maximum statement execution time exceeded
2025-11-17 11:11:58.869 | 	at metabase.driver.sql_jdbc.execute$execute_reducible_query$fn__93479$fn__93480.invoke(execute.clj:770)
2025-11-17 11:11:58.869 | 	at metabase.driver.sql_jdbc.execute$execute_reducible_query$fn__93479.invoke(execute.clj:767)
2025-11-17 11:11:58.869 | 	at metabase.driver.sql_jdbc.execute$fn__93254$fn__93255.invoke(execute.clj:422)
2025-11-17 11:11:58.869 | 	at metabase.driver.sql_jdbc.execute$do_with_resolved_connection.invokeStatic(execute.clj:347)
2025-11-17 11:11:58.869 | 	at metabase.driver.sql_jdbc.execute$do_with_resolved_connection.invoke(execute.clj:327)
2025-11-17 11:11:58.869 | 	at metabase.driver.sql_jdbc.execute$fn__93254.invokeStatic(execute.clj:416)
2025-11-17 11:11:58.869 | 	at metabase.driver.sql_jdbc.execute$fn__93254.invoke(execute.clj:414)
2025-11-17 11:11:58.869 | 	at clojure.lang.MultiFn.invoke(MultiFn.java:244)
2025-11-17 11:11:58.869 | 	at metabase.driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:760)
2025-11-17 11:11:58.869 | 	at metabase.driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:749)
2025-11-17 11:11:58.869 | 	at metabase.driver.sql_jdbc$fn__98159.invokeStatic(sql_jdbc.clj:87)
2025-11-17 11:11:58.869 | 	at metabase.driver.sql_jdbc$fn__98159.invoke(sql_jdbc.clj:85)
2025-11-17 11:11:58.869 | 	at clojure.lang.MultiFn.invoke(MultiFn.java:244)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.pipeline$_STAR_execute_STAR_.invokeStatic(pipeline.clj:59)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.pipeline$_STAR_execute_STAR_.invoke(pipeline.clj:44)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.pipeline$_STAR_run_STAR_.invokeStatic(pipeline.clj:119)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.pipeline$_STAR_run_STAR_.invoke(pipeline.clj:108)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.execute$run.invokeStatic(execute.clj:72)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.execute$run.invoke(execute.clj:62)
2025-11-17 11:11:58.869 | 	at metabase_enterprise.database_routing.middleware$fn__147301$swap_destination_db__147302$fn__147303.invoke(middleware.clj:32)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.middleware.enterprise$swap_destination_db_middleware$fn__78844.invoke(enterprise.clj:75)
2025-11-17 11:11:58.869 | 	at metabase_enterprise.impersonation.middleware$fn__134318$apply_impersonation_postprocessing__134319$fn__134320.invoke(middleware.clj:35)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.middleware.enterprise$apply_impersonation_postprocessing_middleware$fn__78826.invoke(enterprise.clj:50)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.middleware.update_used_cards$update_used_cards_BANG_$fn__80613.invoke(update_used_cards.clj:67)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.execute$add_native_form_to_result_metadata$fn__80653.invoke(execute.clj:27)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.execute$add_preprocessed_query_to_result_metadata_for_userland_query$fn__80660.invoke(execute.clj:39)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.middleware.cache$maybe_return_cached_results$maybe_return_cached_results_STAR___80549.invoke(cache.clj:246)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.middleware.permissions$check_query_permissions$fn__73506.invoke(permissions.clj:153)
2025-11-17 11:11:58.869 | 	at metabase_enterprise.advanced_permissions.query_processor.middleware.permissions$fn__136056$oss_or_ee_fn__47229__auto____136057$fn__136058.invoke(permissions.clj:90)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.middleware.enterprise$check_download_permissions_middleware$fn__78854.invoke(enterprise.clj:92)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.execute$execute$fn__80692.invoke(execute.clj:99)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.setup$do_with_qp_setup.invokeStatic(setup.clj:232)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.setup$do_with_qp_setup.invoke(setup.clj:223)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.execute$execute.invokeStatic(execute.clj:98)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.execute$execute.invoke(execute.clj:94)
2025-11-17 11:11:58.869 | 	at metabase.query_processor$process_query_STAR__STAR_.invokeStatic(query_processor.clj:50)
2025-11-17 11:11:58.869 | 	at metabase.query_processor$process_query_STAR__STAR_.invoke(query_processor.clj:45)
2025-11-17 11:11:58.869 | 	at metabase_enterprise.audit_app.query_processor.middleware.handle_audit_queries$fn__93697$handle_audit_app_internal_queries__93698$fn__93699.invoke(handle_audit_queries.clj:143)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.middleware.enterprise$handle_audit_app_internal_queries_middleware$fn__78882.invoke(enterprise.clj:131)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.middleware.process_userland_query$process_userland_query_middleware$fn__85370.invoke(process_userland_query.clj:172)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.middleware.catch_exceptions$catch_exceptions$fn__85430.invoke(catch_exceptions.clj:132)
2025-11-17 11:11:58.869 | 	at metabase.query_processor$process_query$fn__85470.invoke(query_processor.clj:81)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.setup$do_with_canceled_chan$fn__79952.invoke(setup.clj:196)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.setup$do_with_database_local_settings$fn__79947.invoke(setup.clj:188)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.setup$do_with_driver$fn__79938$fn__79939.invoke(setup.clj:173)
2025-11-17 11:11:58.869 | 	at metabase.driver$do_with_driver.invokeStatic(driver.clj:54)
2025-11-17 11:11:58.869 | 	at metabase.driver$do_with_driver.invoke(driver.clj:49)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.setup$do_with_driver$fn__79938.invoke(setup.clj:172)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.setup$do_with_metadata_provider$fn__79930.invoke(setup.clj:137)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.setup$do_with_resolved_database$fn__79922.invoke(setup.clj:125)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.setup$do_with_qp_setup.invokeStatic(setup.clj:240)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.setup$do_with_qp_setup.invoke(setup.clj:223)
2025-11-17 11:11:58.869 | 	at metabase.query_processor$process_query.invokeStatic(query_processor.clj:79)
2025-11-17 11:11:58.869 | 	at metabase.query_processor$process_query.invoke(query_processor.clj:72)
2025-11-17 11:11:58.869 | 	at metabase.driver.common.table_rows_sample$table_rows_sample.invokeStatic(table_rows_sample.clj:107)
2025-11-17 11:11:58.869 | 	at metabase.driver.common.table_rows_sample$table_rows_sample.invoke(table_rows_sample.clj:82)
2025-11-17 11:11:58.869 | 	at metabase.driver.common.table_rows_sample$fn__93761.invokeStatic(table_rows_sample.clj:111)
2025-11-17 11:11:58.869 | 	at metabase.driver.common.table_rows_sample$fn__93761.invoke(table_rows_sample.clj:109)
2025-11-17 11:11:58.869 | 	at clojure.lang.MultiFn.invoke(MultiFn.java:252)
2025-11-17 11:11:58.869 | 	at metabase.sync.analyze.fingerprint$fingerprint_fields_BANG_.invokeStatic(fingerprint.clj:86)
2025-11-17 11:11:58.869 | 	at metabase.sync.analyze.fingerprint$fingerprint_fields_BANG_.invoke(fingerprint.clj:63)
2025-11-17 11:11:58.869 | 	at metabase.sync.analyze.fingerprint$fingerprint_table_BANG_$fn__70822.invoke(fingerprint.clj:203)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$do_with_returning_throwable.invokeStatic(util.clj:220)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$do_with_returning_throwable.invoke(util.clj:217)
2025-11-17 11:11:58.869 | 	at metabase.sync.analyze.fingerprint$fingerprint_table_BANG_.invokeStatic(fingerprint.clj:202)
2025-11-17 11:11:58.869 | 	at metabase.sync.analyze.fingerprint$fingerprint_table_BANG_.invoke(fingerprint.clj:195)
2025-11-17 11:11:58.869 | 	at metabase.sync.analyze.fingerprint$fingerprint_fields_for_db_BANG__STAR_$fn__70829$fn__70830.invoke(fingerprint.clj:227)
2025-11-17 11:11:58.869 | 	at clojure.core$completing$fn__8562.invoke(core.clj:7010)
2025-11-17 11:11:58.869 | 	at clojure.core$map$fn__5952$fn__5953.invoke(core.clj:2759)
2025-11-17 11:11:58.869 | 	at toucan2.pipeline$with_init$fn__31578.invoke(pipeline.clj:327)
2025-11-17 11:11:58.869 | 	at clojure.core$completing$fn__8562.invoke(core.clj:7010)
2025-11-17 11:11:58.869 | 	at clojure.core$map$fn__5952$fn__5953.invoke(core.clj:2759)
2025-11-17 11:11:58.869 | 	at clojure.core$map$fn__5952$fn__5953.invoke(core.clj:2759)
2025-11-17 11:11:58.869 | 	at toucan2.jdbc.result_set$reduce_result_set.invokeStatic(result_set.clj:156)
2025-11-17 11:11:58.869 | 	at toucan2.jdbc.result_set$reduce_result_set.invoke(result_set.clj:125)
2025-11-17 11:11:58.869 | 	at toucan2.jdbc.query$reduce_jdbc_query.invokeStatic(query.clj:51)
2025-11-17 11:11:58.869 | 	at toucan2.jdbc.query$reduce_jdbc_query.invoke(query.clj:22)
2025-11-17 11:11:58.869 | 	at toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_default_default.invokeStatic(pipeline.clj:19)
2025-11-17 11:11:58.869 | 	at toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_default_default.invoke(pipeline.clj:9)
2025-11-17 11:11:58.869 | 	at methodical.impl.combo.common$partial_STAR_$fn__29022.invoke(common.clj:15)
2025-11-17 11:11:58.869 | 	at methodical.impl.combo.threaded$combine_methods_thread_last$fn__29332$combined_method_thread_last__29333.invoke(threaded.clj:64)
2025-11-17 11:11:58.869 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)
2025-11-17 11:11:58.869 | 	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:65)
2025-11-17 11:11:58.869 | 	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
2025-11-17 11:11:58.869 | 	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:216)
2025-11-17 11:11:58.869 | 	at toucan2.pipeline$transduce_execute$with_connection_STAR___31498.invoke(pipeline.clj:78)
2025-11-17 11:11:58.869 | 	at toucan2.connection$bind_current_connectable_fn$fn__31172.invoke(connection.clj:104)
2025-11-17 11:11:58.869 | 	at toucan2.connection$bind_current_connectable_fn$fn__31172.invoke(connection.clj:104)
2025-11-17 11:11:58.869 | 	at toucan2.connection$bind_current_connectable_fn$fn__31172.invoke(connection.clj:104)
2025-11-17 11:11:58.869 | 	at toucan2.jdbc.connection$do_with_connection_primary_method_javax_sql_DataSource.invokeStatic(connection.clj:18)
2025-11-17 11:11:58.869 | 	at toucan2.jdbc.connection$do_with_connection_primary_method_javax_sql_DataSource.invoke(connection.clj:15)
2025-11-17 11:11:58.869 | 	at methodical.impl.combo.common$partial_STAR_$fn__29022.invoke(common.clj:12)
2025-11-17 11:11:58.869 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)
2025-11-17 11:11:58.869 | 	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)
2025-11-17 11:11:58.869 | 	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)
2025-11-17 11:11:58.869 | 	at methodical.impl.combo.common$partial_STAR_$fn__29022.invoke(common.clj:12)
2025-11-17 11:11:58.869 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)
2025-11-17 11:11:58.869 | 	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)
2025-11-17 11:11:58.869 | 	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
2025-11-17 11:11:58.869 | 	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)
2025-11-17 11:11:58.869 | 	at metabase.app_db.connection$do_with_connection_primary_method_default.invokeStatic(connection.clj:132)
2025-11-17 11:11:58.869 | 	at metabase.app_db.connection$do_with_connection_primary_method_default.invoke(connection.clj:130)
2025-11-17 11:11:58.869 | 	at methodical.impl.combo.common$partial_STAR_$fn__29022.invoke(common.clj:12)
2025-11-17 11:11:58.869 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)
2025-11-17 11:11:58.869 | 	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)
2025-11-17 11:11:58.869 | 	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)
2025-11-17 11:11:58.869 | 	at methodical.impl.combo.common$partial_STAR_$fn__29022.invoke(common.clj:12)
2025-11-17 11:11:58.869 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)
2025-11-17 11:11:58.869 | 	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)
2025-11-17 11:11:58.869 | 	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
2025-11-17 11:11:58.869 | 	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)
2025-11-17 11:11:58.869 | 	at toucan2.connection$do_with_connection_primary_method_.invokeStatic(connection.clj:204)
2025-11-17 11:11:58.869 | 	at toucan2.connection$do_with_connection_primary_method_.invoke(connection.clj:194)
2025-11-17 11:11:58.869 | 	at methodical.impl.combo.common$partial_STAR_$fn__29022.invoke(common.clj:12)
2025-11-17 11:11:58.869 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)
2025-11-17 11:11:58.869 | 	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)
2025-11-17 11:11:58.869 | 	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)
2025-11-17 11:11:58.869 | 	at methodical.impl.combo.common$partial_STAR_$fn__29022.invoke(common.clj:12)
2025-11-17 11:11:58.869 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)
2025-11-17 11:11:58.869 | 	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)
2025-11-17 11:11:58.869 | 	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
2025-11-17 11:11:58.869 | 	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)
2025-11-17 11:11:58.869 | 	at toucan2.pipeline$transduce_execute.invokeStatic(pipeline.clj:77)
2025-11-17 11:11:58.869 | 	at toucan2.pipeline$transduce_execute.invoke(pipeline.clj:64)
2025-11-17 11:11:58.869 | 	at clojure.lang.Var.invoke(Var.java:401)
2025-11-17 11:11:58.869 | 	at toucan2.pipeline$transduce_compiled_query.invokeStatic(pipeline.clj:244)
2025-11-17 11:11:58.869 | 	at toucan2.pipeline$transduce_compiled_query.invoke(pipeline.clj:240)
2025-11-17 11:11:58.869 | 	at toucan2.pipeline$transduce_built_query.invokeStatic(pipeline.clj:252)
2025-11-17 11:11:58.869 | 	at toucan2.pipeline$transduce_built_query.invoke(pipeline.clj:246)
2025-11-17 11:11:58.869 | 	at toucan2.pipeline$transduce_query_primary_method_default.invokeStatic(pipeline.clj:272)
2025-11-17 11:11:58.869 | 	at toucan2.pipeline$transduce_query_primary_method_default.invoke(pipeline.clj:269)
2025-11-17 11:11:58.869 | 	at methodical.impl.combo.common$partial_STAR_$fn__29022.invoke(common.clj:15)
2025-11-17 11:11:58.869 | 	at methodical.impl.combo.threaded$combine_methods_thread_last$fn__29332$combined_method_thread_last__29333.invoke(threaded.clj:64)
2025-11-17 11:11:58.869 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)
2025-11-17 11:11:58.869 | 	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:65)
2025-11-17 11:11:58.869 | 	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
2025-11-17 11:11:58.869 | 	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:216)
2025-11-17 11:11:58.869 | 	at toucan2.pipeline$transduce_query_STAR_.invokeStatic(pipeline.clj:278)
2025-11-17 11:11:58.869 | 	at toucan2.pipeline$transduce_query_STAR_.invoke(pipeline.clj:274)
2025-11-17 11:11:58.869 | 	at toucan2.pipeline$transduce_with_model.invokeStatic(pipeline.clj:293)
2025-11-17 11:11:58.869 | 	at toucan2.pipeline$transduce_with_model.invoke(pipeline.clj:280)
2025-11-17 11:11:58.869 | 	at toucan2.pipeline$transduce_parsed.invokeStatic(pipeline.clj:309)
2025-11-17 11:11:58.869 | 	at toucan2.pipeline$transduce_parsed.invoke(pipeline.clj:295)
2025-11-17 11:11:58.869 | 	at toucan2.pipeline$transduce_unparsed.invokeStatic(pipeline.clj:317)
2025-11-17 11:11:58.869 | 	at toucan2.pipeline$transduce_unparsed.invoke(pipeline.clj:311)
2025-11-17 11:11:58.869 | 	at clojure.lang.AFn.applyToHelper(AFn.java:160)
2025-11-17 11:11:58.869 | 	at clojure.lang.AFn.applyTo(AFn.java:144)
2025-11-17 11:11:58.869 | 	at clojure.core$apply.invokeStatic(core.clj:669)
2025-11-17 11:11:58.869 | 	at clojure.core$apply.invoke(core.clj:662)
2025-11-17 11:11:58.869 | 	at toucan2.pipeline$reducible_fn$reify__31605.reduce(pipeline.clj:404)
2025-11-17 11:11:58.869 | 	at clojure.core$transduce.invokeStatic(core.clj:7025)
2025-11-17 11:11:58.869 | 	at clojure.core.Eduction.reduce(core.clj:7874)
2025-11-17 11:11:58.869 | 	at clojure.core$reduce.invokeStatic(core.clj:6964)
2025-11-17 11:11:58.869 | 	at clojure.core$reduce.invoke(core.clj:6947)
2025-11-17 11:11:58.869 | 	at metabase.sync.analyze.fingerprint$fingerprint_fields_for_db_BANG__STAR_$fn__70829.invoke(fingerprint.clj:225)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.store$do_with_metadata_provider.invokeStatic(store.clj:198)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.store$do_with_metadata_provider.invoke(store.clj:184)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.store$do_with_metadata_provider.invokeStatic(store.clj:191)
2025-11-17 11:11:58.869 | 	at metabase.query_processor.store$do_with_metadata_provider.invoke(store.clj:184)
2025-11-17 11:11:58.869 | 	at metabase.sync.analyze.fingerprint$fingerprint_fields_for_db_BANG__STAR_.invokeStatic(fingerprint.clj:221)
2025-11-17 11:11:58.869 | 	at metabase.sync.analyze.fingerprint$fingerprint_fields_for_db_BANG__STAR_.invoke(fingerprint.clj:212)
2025-11-17 11:11:58.869 | 	at metabase.sync.analyze.fingerprint$fingerprint_fields_for_db_BANG__STAR_.invokeStatic(fingerprint.clj:216)
2025-11-17 11:11:58.869 | 	at metabase.sync.analyze.fingerprint$fingerprint_fields_for_db_BANG__STAR_.invoke(fingerprint.clj:212)
2025-11-17 11:11:58.869 | 	at metabase.sync.analyze.fingerprint$fingerprint_fields_for_db_BANG_.invokeStatic(fingerprint.clj:245)
2025-11-17 11:11:58.869 | 	at metabase.sync.analyze.fingerprint$fingerprint_fields_for_db_BANG_.invoke(fingerprint.clj:240)
2025-11-17 11:11:58.869 | 	at metabase.sync.analyze$make_analyze_steps$fn__70972.invoke(analyze.clj:100)
2025-11-17 11:11:58.869 | 	at clojure.lang.AFn.applyToHelper(AFn.java:154)
2025-11-17 11:11:58.869 | 	at clojure.lang.AFn.applyTo(AFn.java:144)
2025-11-17 11:11:58.869 | 	at clojure.core$apply.invokeStatic(core.clj:669)
2025-11-17 11:11:58.869 | 	at clojure.core$apply.invoke(core.clj:662)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$run_step_with_metadata$fn__66771$fn__66772$fn__66775.invoke(util.clj:531)
2025-11-17 11:11:58.869 | 	at metabase.task_history.models.task_history$do_with_task_history.invokeStatic(task_history.clj:157)
2025-11-17 11:11:58.869 | 	at metabase.task_history.models.task_history$do_with_task_history.invoke(task_history.clj:145)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$run_step_with_metadata$fn__66771$fn__66772.invoke(util.clj:524)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$do_with_returning_throwable.invokeStatic(util.clj:220)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$do_with_returning_throwable.invoke(util.clj:217)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$run_step_with_metadata$fn__66771.doInvoke(util.clj:523)
2025-11-17 11:11:58.869 | 	at clojure.lang.RestFn.invoke(RestFn.java:400)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:134)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:128)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$do_with_start_and_finish_debug_logging.invokeStatic(util.clj:153)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$do_with_start_and_finish_debug_logging.invoke(util.clj:149)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$run_step_with_metadata.invokeStatic(util.clj:518)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$run_step_with_metadata.invoke(util.clj:513)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$run_sync_operation$fn__66816$fn__66824.invoke(util.clj:594)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$run_sync_operation$fn__66816.invoke(util.clj:592)
2025-11-17 11:11:58.869 | 	at metabase.task_history.models.task_history$do_with_task_history.invokeStatic(task_history.clj:157)
2025-11-17 11:11:58.869 | 	at metabase.task_history.models.task_history$do_with_task_history.invoke(task_history.clj:145)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$run_sync_operation.invokeStatic(util.clj:589)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$run_sync_operation.invoke(util.clj:584)
2025-11-17 11:11:58.869 | 	at metabase.sync.analyze$analyze_db_BANG_$fn__70980.invoke(analyze.clj:116)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:198)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$do_with_error_handling.invoke(util.clj:191)
2025-11-17 11:11:58.869 | 	at clojure.core$partial$fn__5931.invoke(core.clj:2647)
2025-11-17 11:11:58.869 | 	at metabase.driver$fn__60380.invokeStatic(driver.clj:993)
2025-11-17 11:11:58.869 | 	at metabase.driver$fn__60380.invoke(driver.clj:993)
2025-11-17 11:11:58.869 | 	at clojure.lang.MultiFn.invoke(MultiFn.java:239)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$sync_in_context$fn__66663.invoke(util.clj:170)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$with_db_logging_disabled$fn__66660.invoke(util.clj:162)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:134)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:128)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$with_start_and_finish_logging$fn__66641.invoke(util.clj:147)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$with_sync_events$fn__66636.invoke(util.clj:120)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$with_duplicate_ops_prevented$fn__66619.invoke(util.clj:92)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$do_sync_operation.invokeStatic(util.clj:241)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$do_sync_operation.invoke(util.clj:234)
2025-11-17 11:11:58.869 | 	at metabase.sync.analyze$analyze_db_BANG_.invokeStatic(analyze.clj:114)
2025-11-17 11:11:58.869 | 	at metabase.sync.analyze$analyze_db_BANG_.invoke(analyze.clj:109)
2025-11-17 11:11:58.869 | 	at metabase.sync.sync$do_phase_BANG_.invokeStatic(sync.clj:49)
2025-11-17 11:11:58.869 | 	at metabase.sync.sync$do_phase_BANG_.invoke(sync.clj:47)
2025-11-17 11:11:58.869 | 	at clojure.core$partial$fn__5929.invoke(core.clj:2641)
2025-11-17 11:11:58.869 | 	at clojure.core$keep$fn__8695.invoke(core.clj:7529)
2025-11-17 11:11:58.869 | 	at clojure.lang.LazySeq.force(LazySeq.java:50)
2025-11-17 11:11:58.869 | 	at clojure.lang.LazySeq.realize(LazySeq.java:89)
2025-11-17 11:11:58.869 | 	at clojure.lang.LazySeq.seq(LazySeq.java:106)
2025-11-17 11:11:58.869 | 	at clojure.lang.RT.seq(RT.java:555)
2025-11-17 11:11:58.869 | 	at clojure.core$seq__5488.invokeStatic(core.clj:139)
2025-11-17 11:11:58.869 | 	at clojure.core$dorun.invokeStatic(core.clj:3141)
2025-11-17 11:11:58.869 | 	at clojure.core$doall.invokeStatic(core.clj:3156)
2025-11-17 11:11:58.869 | 	at clojure.core$doall.invoke(core.clj:3156)
2025-11-17 11:11:58.869 | 	at metabase.sync.sync$sync_database_BANG_$fn__71152.invoke(sync.clj:74)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:198)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$do_with_error_handling.invoke(util.clj:191)
2025-11-17 11:11:58.869 | 	at clojure.core$partial$fn__5931.invoke(core.clj:2647)
2025-11-17 11:11:58.869 | 	at metabase.driver$fn__60380.invokeStatic(driver.clj:993)
2025-11-17 11:11:58.869 | 	at metabase.driver$fn__60380.invoke(driver.clj:993)
2025-11-17 11:11:58.869 | 	at clojure.lang.MultiFn.invoke(MultiFn.java:239)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$sync_in_context$fn__66663.invoke(util.clj:170)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$with_db_logging_disabled$fn__66660.invoke(util.clj:162)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:134)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:128)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$with_start_and_finish_logging$fn__66641.invoke(util.clj:147)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$with_sync_events$fn__66636.invoke(util.clj:120)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$with_duplicate_ops_prevented$fn__66619.invoke(util.clj:92)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$do_sync_operation.invokeStatic(util.clj:241)
2025-11-17 11:11:58.869 | 	at metabase.sync.util$do_sync_operation.invoke(util.clj:234)
2025-11-17 11:11:58.869 | 	at metabase.sync.sync$sync_database_BANG_.invokeStatic(sync.clj:71)
2025-11-17 11:11:58.869 | 	at metabase.sync.sync$sync_database_BANG_.invoke(sync.clj:57)
2025-11-17 11:11:58.869 | 	at metabase.sync.sync$sync_database_BANG_.invokeStatic(sync.clj:66)
2025-11-17 11:11:58.869 | 	at metabase.sync.sync$sync_database_BANG_.invoke(sync.clj:57)
2025-11-17 11:11:58.869 | 	at metabase.sync.events.sync_database$publish_event_BANG__primary_method_metabase_sync_events_sync_database_event$fn__136156.invoke(sync_database.clj:26)
2025-11-17 11:11:58.869 | 	at clojure.lang.AFn.applyToHelper(AFn.java:152)
2025-11-17 11:11:58.869 | 	at clojure.lang.AFn.applyTo(AFn.java:144)
2025-11-17 11:11:58.869 | 	at clojure.core$apply.invokeStatic(core.clj:667)
2025-11-17 11:11:58.869 | 	at clojure.core$with_bindings_STAR_.invokeStatic(core.clj:1990)
2025-11-17 11:11:58.869 | 	at clojure.core$with_bindings_STAR_.doInvoke(core.clj:1990)
2025-11-17 11:11:58.869 | 	at clojure.lang.RestFn.invoke(RestFn.java:428)
2025-11-17 11:11:58.869 | 	at clojure.lang.AFn.applyToHelper(AFn.java:156)
2025-11-17 11:11:58.869 | 	at clojure.lang.RestFn.applyTo(RestFn.java:135)
2025-11-17 11:11:58.869 | 	at clojure.core$apply.invokeStatic(core.clj:671)
2025-11-17 11:11:58.869 | 	at clojure.core$bound_fn_STAR_$fn__5839.doInvoke(core.clj:2020)
2025-11-17 11:11:58.869 | 	at clojure.lang.RestFn.invoke(RestFn.java:400)
2025-11-17 11:11:58.869 | 	at clojure.lang.AFn.call(AFn.java:18)
2025-11-17 11:11:58.869 | 	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
2025-11-17 11:11:58.869 | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
2025-11-17 11:11:58.869 | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
2025-11-17 11:11:58.869 | 	at java.base/java.lang.Thread.run(Unknown Source)
2025-11-17 11:11:58.869 | Caused by: java.sql.SQLTransientConnectionException: (conn=440) Query execution was interrupted, maximum statement execution time exceeded
2025-11-17 11:11:58.869 | 	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:79)
2025-11-17 11:11:58.869 | 	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.create(ExceptionFactory.java:158)
2025-11-17 11:11:58.869 | 	at org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(MariaDbStatement.java:262)
2025-11-17 11:11:58.869 | 	at org.mariadb.jdbc.MariaDbStatement.executeInternal(MariaDbStatement.java:362)
2025-11-17 11:11:58.869 | 	at org.mariadb.jdbc.MariaDbStatement.execute(MariaDbStatement.java:500)
2025-11-17 11:11:58.869 | 	at com.mchange.v2.c3p0.impl.NewProxyStatement.execute(NewProxyStatement.java:1341)
2025-11-17 11:11:58.869 | 	at metabase.driver.sql_jdbc.execute$fn__93385.invokeStatic(execute.clj:582)
2025-11-17 11:11:58.869 | 	at metabase.driver.sql_jdbc.execute$fn__93385.invoke(execute.clj:580)
2025-11-17 11:11:58.869 | 	at clojure.lang.MultiFn.invoke(MultiFn.java:239)
2025-11-17 11:11:58.869 | 	at metabase.driver.sql_jdbc.execute$execute_statement_or_prepared_statement_BANG_.invokeStatic(execute.clj:590)
2025-11-17 11:11:58.869 | 	at metabase.driver.sql_jdbc.execute$execute_statement_or_prepared_statement_BANG_.invoke(execute.clj:587)
2025-11-17 11:11:58.869 | 	at metabase.driver.sql_jdbc.execute$execute_reducible_query$fn__93479$fn__93480.invoke(execute.clj:768)
2025-11-17 11:11:58.869 | 	... 256 more
2025-11-17 11:11:58.869 | Caused by: org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException: Query execution was interrupted, maximum statement execution time exceeded
2025-11-17 11:11:58.869 | 	at org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException.of(MariaDbSqlException.java:34)
2025-11-17 11:11:58.869 | 	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:195)
2025-11-17 11:11:58.869 | 	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:263)
2025-11-17 11:11:58.869 | 	at org.mariadb.jdbc.MariaDbStatement.executeInternal(MariaDbStatement.java:356)
2025-11-17 11:11:58.869 | 	... 264 more
2025-11-17 11:11:58.869 | Caused by: java.sql.SQLException: Query execution was interrupted, maximum statement execution time exceeded
2025-11-17 11:11:58.869 | 	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(AbstractQueryProtocol.java:1693)
2025-11-17 11:11:58.869 | 	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(AbstractQueryProtocol.java:1555)
2025-11-17 11:11:58.869 | 	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:1518)
2025-11-17 11:11:58.869 | 	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:257)
2025-11-17 11:11:58.869 | 	... 265 more
2025-11-17 11:11:58.871 | 2025-11-17 14:11:58,871 INFO sync.util :: FINISHED: step ''fingerprint-fields'' for mysql Database 2 ''mysql-14020'' (1.1 mins) (free: 0.53G, total: 0.84G, ratio: 0.62) 
2025-11-17 11:11:58.872 | 2025-11-17 14:11:58,872 INFO sync.util :: STARTING: step ''classify-fields'' for mysql Database 2 ''mysql-14020'' (free: 0.53G, total: 0.84G, ratio: 0.62) 
2025-11-17 11:11:58.877 | 2025-11-17 14:11:58,877 INFO sync.analyze :: classify-fields Analyzed [*************************·························] 😬   50% Table 142 ''my_sleep_view'' 
2025-11-17 11:11:58.879 | 2025-11-17 14:11:58,878 INFO sync.util :: FINISHED: step ''classify-fields'' for mysql Database 2 ''mysql-14020'' (6.9 ms) (free: 0.52G, total: 0.84G, ratio: 0.62) 
2025-11-17 11:11:58.879 | 2025-11-17 14:11:58,879 INFO sync.util :: STARTING: step ''classify-tables'' for mysql Database 2 ''mysql-14020'' (free: 0.52G, total: 0.84G, ratio: 0.62) 
2025-11-17 11:11:58.883 | 2025-11-17 14:11:58,883 INFO sync.analyze :: classify-tables Analyzed [*************************************·············] 😋   75% Table 142 ''my_sleep_view'' 
2025-11-17 11:11:58.884 | 2025-11-17 14:11:58,884 INFO sync.util :: FINISHED: step ''classify-tables'' for mysql Database 2 ''mysql-14020'' (5.1 ms) (free: 0.52G, total: 0.84G, ratio: 0.61) 
2025-11-17 11:11:58.885 | 2025-11-17 14:11:58,885 INFO sync.util :: FINISHED: Analyze data for mysql Database 2 ''mysql-14020'' (1.1 mins) (free: 0.52G, total: 0.84G, ratio: 0.61) 
2025-11-17 11:11:58.885 | 2025-11-17 14:11:58,885 INFO sync.util :: STARTING: Cache field values in mysql Database 2 ''mysql-14020'' (free: 0.52G, total: 0.84G, ratio: 0.61) 
2025-11-17 11:11:58.886 | 2025-11-17 14:11:58,886 INFO sync.util :: STARTING: step ''delete-expired-advanced-field-values'' for mysql Database 2 ''mysql-14020'' (free: 0.52G, total: 0.84G, ratio: 0.61) 
2025-11-17 11:11:58.896 | 2025-11-17 14:11:58,896 INFO sync.util :: FINISHED: step ''delete-expired-advanced-field-values'' for mysql Database 2 ''mysql-14020'' (9.9 ms) (free: 0.50G, total: 0.84G, ratio: 0.60) 
2025-11-17 11:11:58.896 | 2025-11-17 14:11:58,896 INFO sync.util :: STARTING: step ''update-field-values'' for mysql Database 2 ''mysql-14020'' (free: 0.50G, total: 0.84G, ratio: 0.60) 
2025-11-17 11:11:58.898 | 2025-11-17 14:11:58,898 INFO sync.field-values :: Field Field 1261 ''my_col'' does not have FieldValues. Skipping... 
2025-11-17 11:11:58.899 | 2025-11-17 14:11:58,899 INFO sync.util :: FINISHED: step ''update-field-values'' for mysql Database 2 ''mysql-14020'' (2.7 ms) (free: 0.50G, total: 0.84G, ratio: 0.60) 
2025-11-17 11:11:58.899 | 2025-11-17 14:11:58,899 INFO sync.util :: FINISHED: Cache field values in mysql Database 2 ''mysql-14020'' (14.2 ms) (free: 0.50G, total: 0.84G, ratio: 0.60) 
2025-11-17 11:11:58.900 | 2025-11-17 14:11:58,899 INFO sync.util :: FINISHED: Sync mysql Database 2 ''mysql-14020'' (1.3 mins) (free: 0.50G, total: 0.84G, ratio: 0.60)
```

### Information about your Metabase installation

```JSON
v57.2
```

### Severity

P2

### Additional context

Workarounds:

* hide the slow view from Table Metadata with the eye icon so it's skipped when fingerprinting
* workaround from the underlying MySQL timeout issue [https://github.com/metabase/metabase/issues/59070#issuecomment-2947347515](<https://github.com/metabase/metabase/issues/59070#issuecomment-2947347515>)

For the [customer](<https://app.usepylon.com/issues/views/9563cff7-93de-4f1b-8598-cafd59f31700?conversationID=d712bd64-dd1e-4a85-ad86-5fa1489a5e11>) that reported this, these queries started accumulating over days, reaching the 2h server timeout after which the connection was reset, and then retrying every sync

--- a/src/metabase/query_processor/setup.clj
+++ b/src/metabase/query_processor/setup.clj
@@ -5,6 +5,7 @@
    [clojure.set :as set]
    [metabase.config.core :as config]
    [metabase.driver :as driver]
+   [metabase.driver.settings :as driver.settings]
    [metabase.driver.util :as driver.u]
    [metabase.lib.computed :as lib.computed]
    [metabase.lib.metadata :as lib.metadata]
@@ -187,12 +188,29 @@
         (setting/with-database db
           (f query))))))
 
+(defn- schedule-query-timeout-cancel!
+  "Schedule a put to `canceled-chan` after `*query-timeout-ms*`. Driver execution wires this channel to
+  `Statement.cancel()` (see [[metabase.driver.sql-jdbc.execute/wire-up-canceled-chan-to-cancel-Statement!]]), which
+  for MySQL/MariaDB issues `KILL QUERY` on a side connection — so this is the cross-driver path that actually stops
+  a running query on the server when the timeout fires. If the query finishes first the channel closes, the
+  `alts!!` returns its value, and the timeout branch is never taken."
+  [canceled-chan]
+  (let [timeout-ms (long driver.settings/*query-timeout-ms*)]
+    (a/go
+      (let [timeout-ch (a/timeout timeout-ms)
+            [_ port]   (a/alts! [canceled-chan timeout-ch])]
+        (when (= port timeout-ch)
+          (log/warnf "Query exceeded timeout of %d ms; canceling" timeout-ms)
+          (a/>! canceled-chan ::timeout))))))
+
 (mu/defn- do-with-canceled-chan :- fn?
   [f :- [:=> [:cat ::qp.schema/any-query] :any]]
   (fn [query]
     (if qp.pipeline/*canceled-chan*
-      (f query)
+      (do (schedule-query-timeout-cancel! qp.pipeline/*canceled-chan*)
+          (f query))
       (binding [qp.pipeline/*canceled-chan* (a/promise-chan)]
+        (schedule-query-timeout-cancel! qp.pipeline/*canceled-chan*)
         (f query)))))
 
 (def ^:private setup-middleware

--- a/src/metabase/query_processor/setup.clj
+++ b/src/metabase/query_processor/setup.clj
@@ -209,12 +209,9 @@
   (fn [query]
     (let [done-ch (a/promise-chan)]
       (try
-        (if qp.pipeline/*canceled-chan*
-          (do (schedule-query-timeout-cancel! qp.pipeline/*canceled-chan* done-ch)
-              (f query))
-          (binding [qp.pipeline/*canceled-chan* (a/promise-chan)]
-            (schedule-query-timeout-cancel! qp.pipeline/*canceled-chan* done-ch)
-            (f query)))
+        (binding [qp.pipeline/*canceled-chan* (or qp.pipeline/*canceled-chan* (a/promise-chan))]
+          (schedule-query-timeout-cancel! qp.pipeline/*canceled-chan* done-ch)
+          (f query))
         (finally
           (a/put! done-ch ::query-finished))))))
 

--- a/src/metabase/query_processor/setup.clj
+++ b/src/metabase/query_processor/setup.clj
@@ -200,7 +200,7 @@
     (a/go
       (let [timeout-ch (a/timeout timeout-ms)
             [_ port]   (a/alts! [done-ch timeout-ch])]
-        (when (= port timeout-ch)
+        (when (identical? port timeout-ch)
           (log/warnf "Query exceeded timeout of %d ms; canceling" timeout-ms)
           (a/put! canceled-chan ::timeout))))))
 
@@ -216,7 +216,7 @@
             (schedule-query-timeout-cancel! qp.pipeline/*canceled-chan* done-ch)
             (f query)))
         (finally
-          (a/close! done-ch))))))
+          (a/put! done-ch ::query-finished))))))
 
 (def ^:private setup-middleware
   "Setup middleware has the signature

--- a/src/metabase/query_processor/setup.clj
+++ b/src/metabase/query_processor/setup.clj
@@ -189,29 +189,34 @@
           (f query))))))
 
 (defn- schedule-query-timeout-cancel!
-  "Schedule a put to `canceled-chan` after `*query-timeout-ms*`. Driver execution wires this channel to
-  `Statement.cancel()` (see [[metabase.driver.sql-jdbc.execute/wire-up-canceled-chan-to-cancel-Statement!]]), which
-  for MySQL/MariaDB issues `KILL QUERY` on a side connection — so this is the cross-driver path that actually stops
-  a running query on the server when the timeout fires. If the query finishes first the channel closes, the
-  `alts!!` returns its value, and the timeout branch is never taken."
-  [canceled-chan]
+  "Schedule a put of `::timeout` to `canceled-chan` after `*query-timeout-ms*`. Driver execution wires this channel
+  to `Statement.cancel()` (see [[metabase.driver.sql-jdbc.execute/wire-up-canceled-chan-to-cancel-Statement!]]),
+  which for MySQL/MariaDB issues `KILL QUERY` on a side connection — so this is the cross-driver path that actually
+  stops a running query on the server when the timeout fires. `done-ch` closes when the query completes, releasing
+  the go-block early. The timer never reads from `canceled-chan`, since callers (notably tests) may bind a regular
+  channel where `alts!`/`<!` would consume the cancel signal away from the driver's listener."
+  [canceled-chan done-ch]
   (let [timeout-ms (long driver.settings/*query-timeout-ms*)]
     (a/go
       (let [timeout-ch (a/timeout timeout-ms)
-            [_ port]   (a/alts! [canceled-chan timeout-ch])]
+            [_ port]   (a/alts! [done-ch timeout-ch])]
         (when (= port timeout-ch)
           (log/warnf "Query exceeded timeout of %d ms; canceling" timeout-ms)
-          (a/>! canceled-chan ::timeout))))))
+          (a/put! canceled-chan ::timeout))))))
 
 (mu/defn- do-with-canceled-chan :- fn?
   [f :- [:=> [:cat ::qp.schema/any-query] :any]]
   (fn [query]
-    (if qp.pipeline/*canceled-chan*
-      (do (schedule-query-timeout-cancel! qp.pipeline/*canceled-chan*)
-          (f query))
-      (binding [qp.pipeline/*canceled-chan* (a/promise-chan)]
-        (schedule-query-timeout-cancel! qp.pipeline/*canceled-chan*)
-        (f query)))))
+    (let [done-ch (a/promise-chan)]
+      (try
+        (if qp.pipeline/*canceled-chan*
+          (do (schedule-query-timeout-cancel! qp.pipeline/*canceled-chan* done-ch)
+              (f query))
+          (binding [qp.pipeline/*canceled-chan* (a/promise-chan)]
+            (schedule-query-timeout-cancel! qp.pipeline/*canceled-chan* done-ch)
+            (f query)))
+        (finally
+          (a/close! done-ch))))))
 
 (def ^:private setup-middleware
   "Setup middleware has the signature

--- a/test/metabase/driver/mysql_test.clj
+++ b/test/metabase/driver/mysql_test.clj
@@ -15,6 +15,7 @@
    [metabase.driver.mysql :as mysql]
    [metabase.driver.mysql.actions :as mysql.actions]
    [metabase.driver.mysql.ddl :as mysql.ddl]
+   [metabase.driver.settings :as driver.settings]
    [metabase.driver.sql-jdbc :as driver.sql-jdbc]
    [metabase.driver.sql-jdbc.actions :as sql-jdbc.actions]
    [metabase.driver.sql-jdbc.actions-test :as sql-jdbc.actions-test]
@@ -1077,3 +1078,24 @@
 
          "webapp@localhost"
          "SET ROLE 'webapp'@'localhost';")))))
+
+(deftest ^:synchronized cancel-slow-mysql-query-via-query-timeout-test
+  (mt/test-driver :mysql
+    (testing "Slow MySQL query is cancelled server-side when *query-timeout-ms* elapses (GHY-3266)"
+      ;; `SELECT SLEEP(60)` would normally take 60s. With `*query-timeout-ms*` of 2s the canceled-chan timer fires
+      ;; (and `Statement.setQueryTimeout` also fires); both reach the MariaDB Connector/J `.cancel()` path which
+      ;; issues `KILL QUERY` on a side connection. The query should error within a few seconds, not run to
+      ;; completion. Cancellation evidence: the query throws AND finishes far short of 60s. We don't assert on the
+      ;; exception message because QP output-schema middleware wraps the underlying SQLException — the timing is
+      ;; the load-bearing assertion.
+      (binding [driver.settings/*query-timeout-ms* 2000]
+        (let [timer   (u/start-timer)
+              query   (mt/native-query {:query "SELECT SLEEP(60) AS s"})
+              result  (try
+                        (qp/process-query query)
+                        (catch Throwable e e))
+              elapsed (u/since-ms timer)]
+          (is (instance? Throwable result)
+              "query should throw rather than completing normally")
+          (is (< elapsed 30000)
+              (format "query should be cancelled well before SLEEP(60) completes naturally — took %.0f ms" elapsed)))))))

--- a/test/metabase/query_processor/setup_test.clj
+++ b/test/metabase/query_processor/setup_test.clj
@@ -60,20 +60,17 @@
     ;; driver listener for the value and on a regular chan one of them consumes it, breaking the other.
     (binding [driver.settings/*query-timeout-ms* 60000]
       (let [external-chan (a/chan)
-            seen          (promise)
             slow-f        (fn [_query]
                             (a/>!! qp.pipeline/*canceled-chan* ::external-cancel)
                             ;; give the timer's go-block a moment to (incorrectly) consume the message
-                            (Thread/sleep 100)
-                            (deliver seen (a/<!! (a/timeout 200))))]
+                            (Thread/sleep 300))]
         (binding [qp.pipeline/*canceled-chan* external-chan]
           (let [f (#'qp.setup/do-with-canceled-chan slow-f)
                 ;; consume from the external chan on a separate thread, mimicking what a driver's cancel listener
                 ;; does. If the timer consumed the message, this take will time out.
                 taken (future (a/alts!! [external-chan (a/timeout 1000)]))]
             (f {:type :internal})
-            @seen
-            (let [[v p] @taken]
+            (let [[v p] (deref taken 2000 [::deref-timed-out ::deref-timed-out])]
               (is (identical? external-chan p)
                   "the external listener should win the take, not time out")
               (is (= ::external-cancel v)

--- a/test/metabase/query_processor/setup_test.clj
+++ b/test/metabase/query_processor/setup_test.clj
@@ -1,10 +1,53 @@
 (ns metabase.query-processor.setup-test
   (:require
+   [clojure.core.async :as a]
    [clojure.test :refer :all]
+   [metabase.driver.settings :as driver.settings]
+   [metabase.query-processor.pipeline :as qp.pipeline]
    [metabase.query-processor.setup :as qp.setup]))
+
+(set! *warn-on-reflection* true)
 
 (deftest ^:parallel internal-query-type-test
   (testing "Make sure internal (audit app) queries work, even tho they don't have a :database ID."
     (qp.setup/with-qp-setup [query {:type :internal}]
       (is (= {:type :internal}
              query)))))
+
+(deftest ^:parallel canceled-chan-timer-fires-on-query-timeout-test
+  (testing "When a query runs longer than *query-timeout-ms*, *canceled-chan* receives a ::timeout message"
+    ;; this is the cross-driver cancel path that backs up Statement.setQueryTimeout — see GHY-3266 (MySQL slow view
+    ;; fingerprinting): when setQueryTimeout doesn't reliably KILL QUERY on the server, the timer here puts
+    ;; ::timeout on *canceled-chan*, which drives Statement.cancel() and forces the driver to issue a server-side
+    ;; cancel anyway.
+    (binding [driver.settings/*query-timeout-ms* 100]
+      (let [observed-chan (promise)
+            observed-val  (promise)
+            slow-f        (fn [_query]
+                            (let [chan qp.pipeline/*canceled-chan*]
+                              (deliver observed-chan chan)
+                              (let [[v p] (a/alts!! [chan (a/timeout 2000)])]
+                                (deliver observed-val {:val v :from-canceled-chan? (identical? p chan)}))))
+            f             (#'qp.setup/do-with-canceled-chan slow-f)]
+        (f {:type :internal})
+        (is (some? @observed-chan)
+            "the QP should bind a *canceled-chan* for the query")
+        (is (:from-canceled-chan? @observed-val)
+            "the canceled-chan should have been signaled before the 2s fallback timeout")
+        (is (= ::qp.setup/timeout (:val @observed-val))
+            "the timer should put ::timeout to *canceled-chan* once *query-timeout-ms* elapses")))))
+
+(deftest ^:parallel canceled-chan-timer-does-not-fire-when-query-completes-first-test
+  (testing "When a query finishes before *query-timeout-ms*, the timer never puts ::timeout on the chan"
+    (binding [driver.settings/*query-timeout-ms* 60000]
+      (let [observed-after-close (promise)
+            fast-f               (fn [_query]
+                                   ;; mimic the pipeline's success path which closes *canceled-chan*
+                                   (a/close! qp.pipeline/*canceled-chan*)
+                                   ;; give the go-block a moment to see the close
+                                   (Thread/sleep 100)
+                                   (deliver observed-after-close (a/poll! qp.pipeline/*canceled-chan*)))
+            f                    (#'qp.setup/do-with-canceled-chan fast-f)]
+        (f {:type :internal})
+        (is (nil? @observed-after-close)
+            "a closed chan should poll nil — the timer must not have written ::timeout to it")))))

--- a/test/metabase/query_processor/setup_test.clj
+++ b/test/metabase/query_processor/setup_test.clj
@@ -51,3 +51,30 @@
         (f {:type :internal})
         (is (nil? @observed-after-close)
             "a closed chan should poll nil — the timer must not have written ::timeout to it")))))
+
+(deftest ^:parallel canceled-chan-timer-does-not-consume-external-cancel-signal-test
+  (testing "Timer must not consume the cancel signal when callers bind a regular (a/chan)"
+    ;; Regression for mongo `kill-an-in-flight-query-test`: that test binds `*canceled-chan*` to a regular
+    ;; `(a/chan)` (not a promise-chan) and puts a cancel sentinel on it; the driver listener takes from the chan
+    ;; to abort the in-flight query. If the timer reads from `*canceled-chan*` (via `alts!`/`<!`), it races the
+    ;; driver listener for the value and on a regular chan one of them consumes it, breaking the other.
+    (binding [driver.settings/*query-timeout-ms* 60000]
+      (let [external-chan (a/chan)
+            seen          (promise)
+            slow-f        (fn [_query]
+                            (a/>!! qp.pipeline/*canceled-chan* ::external-cancel)
+                            ;; give the timer's go-block a moment to (incorrectly) consume the message
+                            (Thread/sleep 100)
+                            (deliver seen (a/<!! (a/timeout 200))))]
+        (binding [qp.pipeline/*canceled-chan* external-chan]
+          (let [f (#'qp.setup/do-with-canceled-chan slow-f)
+                ;; consume from the external chan on a separate thread, mimicking what a driver's cancel listener
+                ;; does. If the timer consumed the message, this take will time out.
+                taken (future (a/alts!! [external-chan (a/timeout 1000)]))]
+            (f {:type :internal})
+            @seen
+            (let [[v p] @taken]
+              (is (identical? external-chan p)
+                  "the external listener should win the take, not time out")
+              (is (= ::external-cancel v)
+                  "the external listener should see the original cancel sentinel"))))))))

